### PR TITLE
add requirements for building olena (g+= make libtool autoconf)

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,8 +58,9 @@ endif
 	cd $(OLENA_DIR) && autoreconf -i
 
 deps-ubuntu:
-	apt-get -y install libmagick++-dev libgraphicsmagick++1-dev libboost-dev \
-		xmlstarlet ca-certificates
+	apt-get -y install \
+		g++ make libtool autoconf \
```
Comment (bertsky) [inline Makefile]: I already have a list of packages required at build time [in Dockerfile](https://github.com/OCR-D/ocrd_olena/blob/3262fee8c180724fec56b1c69c3d686b97d47084/Dockerfile#L19): `g++ make automake git`.

You need `git` for (the default) `OLENA_VERSION=git`.

And `automake` implies `autoconf`, which implies `libtool`.

**Review by bertsky — CHANGES_REQUESTED** [review]
Yes, we should add the build-time requirements here as well.

File: Makefile (hunk 1 above)
Comment (kba) [reply Makefile]: I see, thanks. We're following/adapting the setup guide which already installed git, so I did not (but should) list it here as well. libtool < autoconf < automake is good to know too.

**Comment by kba** [comment]
Line 21 of the Dockerfile

```dockerfile
 apt-get -y install --no-install-recommends $DEPS && \
```

is redundant now but it does not hurt IMHO and makes it clearer what the build deps are.

**Comment by bertsky** [comment]
> Line 21 of the Dockerfile
> 
> ```dockerfile
>  apt-get -y install --no-install-recommends $DEPS && \
> ```
> 
> is redundant now but it does not hurt IMHO and makes it clearer what the build deps are.

I concur. Let's keep it like that.

**Review by bertsky — APPROVED** [review]

**Pull request merged**